src/main/java/org/apache/commons/configuration2/convert/PropertyConverter.java

```diff
@@ -607,12 +612,22 @@ static Object toInternetAddress(final Object value) throws ConversionException {
         if (value.getClass().getName().equals(INTERNET_ADDRESS_CLASSNAME)) {
             return value;
         }
+        if (value.getClass().getName().equals(INTERNET_ADDRESS_CLASSNAME_JAKARTA)) {
+            return value;
+        }
         if (!(value instanceof String)) {
             throw new ConversionException("The value " + value + " can't be converted to a InternetAddress");
         }
         try {
-            final Constructor<?> ctor = Class.forName(INTERNET_ADDRESS_CLASSNAME).getConstructor(String.class);
-            return ctor.newInstance(value);
+            try {
+                // javamail-2.0+, with jakarta.mail.* namespace.
+                final Constructor<?> ctor = Class.forName(INTERNET_ADDRESS_CLASSNAME_JAKARTA).getConstructor(String.class);
+                return ctor.newInstance(value);
+            } catch (ClassNotFoundException e) {
+                // maybe javamail-1.*? With javax.mail.* namespace.
```

Comment:
I am not a fan of trying to support multiple versions of the same API here or anywhere in the code base.
We support one jar or the other. This seems like the same issue @kinow raised in https://github.com/apache/commons-configuration/pull/107/files#r868840083

Reply:
So it sounds like we should just drop the javax API version. How would this change affect users upgrading from the previous version?

Reply:
I vote we just go ahead and update to the latest here. Any objections?

Reply:
None from me :+1: feel free to take over the PR if you'd like/have time :)